firmware/lm32/heartbeat.c

```diff
@@ -0,0 +1,83 @@
+
+#include <generated/csr.h>
+#include <generated/mem.h>
+#include <hw/flags.h>
+#include <system.h>
+#include <time.h>
+
+#include "heartbeat.h"
+#include "processor.h"
+#include "hdmi_in0.h"
+#include "hdmi_in1.h"
+#include "pattern.h"
+
+static int heartbeat_status = 0;
+
+void hb_status(int val)
+{
+	if(val==1) { 
+		heartbeat_status = 1; 
+	}
+	else {
+		heartbeat_status = 0;
+	}		
+}
+
+void hb_service(int sink)
+{
+	static int last_event;
+	static int counter;		
+	static bool color_v;
+	if (heartbeat_status==1) {	
+		if(elapsed(&last_event, identifier_frequency_read()/120)) {
```

Comment:
Increasing this number seemed to be fixing the glitch. Which I suppose was because, we were not filling frames fast enough. Assuming a 60Hz output, setting this to 120, means we are filling frames at double the input rate. Any ways I have added the required definitions now.